### ISSUE TITLE
Several functions (Subscribe, Connect, ect...) never return if there is an error

### ISSUE THREAD

**Issue by LorenzGardner** [noble/noble#830]
calling subscribe on a characteristic with a callback never returns in the case of certain errors. For instance if the characteristic in question doesn't support subscriptions. No error, no callback, just silent failure. The same seems to be true if the actual packet sent is dropped by the peripheral.

Similar problems occur on connect, write, ect... 

Work around seems to be to wrap all nobel calls in setTimeout functions and retry/reconnect as needed. But that is quite painful, and if you miss one your app will just hang. It really seems that the callbacks should happen, especially as the callback take an error param to report such things.

(This is on windows 10 and Node 8 btw).

**Comment by wicktus** [comment]
Using tools like HCI dump, do you see any errors ?

**Comment by LorenzGardner** [comment]
Yes, of course. The HCI stuff is returning errors, but they aren't getting bubbled up (hence what I said in the OP).
